builder/simplify_rules.h

```diff
@@ -512,6 +512,12 @@ bool apply_less_rules(Fn&& apply) {
       apply(c0 < x/c1, 
         (c0 + 1)*c1 <= x, c1 > 0,
         c0*c1 > x, c1 < 0) ||
+      apply(x/may_be<1>(c3) < x/may_be<1>(c3) + c2, true, c2 > 0 && c3 > 0) ||
+      apply(x/may_be<1>(c3) < x/may_be<1>(c3) + c2, false, c2 < 0 && c3 > 0) ||
+      apply((max(x, c0)/may_be<1>(c3)) < max(x, c1)/may_be<1>(c3) + c2, false, c2 < 0 && c3 > 0 && c1/c3 + c2 <= c0/c3) ||
+      apply((max(x, c0)/may_be<1>(c3)) < max(x, c1)/may_be<1>(c3) + c2, true, c2 > 0 && c3 > 0 && c1/c3 + c2 > c0/c3) ||
+      apply((min(x, c0)/may_be<1>(c3)) < min(x, c1)/may_be<1>(c3) + c2, false, c2 < 0 && c3 > 0 && c1/c3 + c2 <= c0/c3) ||
+      apply((min(x, c0)/may_be<1>(c3)) < min(x, c1)/may_be<1>(c3) + c2, true, c2 > 0 && c3 > 0 && c1/c3 + c2 > c0/c3) ||
```

Comment:
These two (pairs) can be rewritten to be 2 rules (one for each pair), using the above mentioned pattern.

I'm wondering, I know I said you need c2 > 0 or c2 < 0, but do you? I'm not sure. The easiest way to find out would be to just remove it, and let the rule fuzzer tell you (run it with `--test_filter=fuzz_rules* --runs_per_test=1000`)

If you don't need it, then I'm wondering, maybe these rules should just be:
```
apply((min(x, y)/may_be<1>(c0)) < min(x, z)/may_be<1>(c0) + c1, y/c0 < z/c0 + c1), c0 > 0) ||
```

It seems like maybe these rules are basically the same as line 537 but with some added constants (the denominator and the added constant)?